internal/cucumber/config.go

```diff
@@ -240,6 +243,24 @@ func shardSuites(rootDir string, suites []Suite, ccy int) ([]Suite, error) {
 			msg.SuiteSplitNoMatch(s.Name, rootDir, s.Options.Paths)
 			return []Suite{}, fmt.Errorf("suite '%s' patterns have no matching files", s.Name)
 		}
+
+		if s.ShardGrepEnabled && len(s.Options.Tags) > 0 {
+			tags := make([]string, len(s.Options.Tags))
+			for i, t := range s.Options.Tags {
+				tags[i] = fmt.Sprintf("(%s)", t)
+			}
+			tagExp := strings.Join(tags, " and ")
+
+			var unmatched []string
+			files, unmatched = tag.MatchFiles(os.DirFS(rootDir), files, tagExp)
+
+			if len(files) == 0 {
+				log.Error().Str("suiteName", s.Name).Str("tagExpression", tagExp).Msg("No files match the configured tagExpressions")
+			} else if len(unmatched) > 0 {
+				log.Info().Str("suiteName", s.Name).Str("tagExpression", tagExp).Msgf("Files filtered out by tagExpression: [%s]", unmatched)
+			}
```

Comment:
(style) Just to keep those extra long lines a wee bit in check:
```suggestion
			if len(files) == 0 {
				log.Error().
					Str("suiteName", s.Name).
					Str("tagExpression", tagExp).
					Msg("No files match the configured tagExpressions")
			} else if len(unmatched) > 0 {
				log.Info().Str("suiteName", s.Name).
					Str("tagExpression", tagExp).
					Msgf("Files filtered out by tagExpression: [%s]", unmatched)
			}
```